app/uk/gov/hmrc/tariffclassificationfrontend/forms/LiabilityFormData.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tariffclassificationfrontend.forms
+
+import java.time.ZoneOffset._
+import java.time.{Instant, LocalDate}
+
+import play.api.data.Form
+import play.api.data.Forms._
+import uk.gov.hmrc.tariffclassificationfrontend.forms.mappings.FormMappings
+
+import scala.util.Try
+
+
+case class LiabilityFormData(entryDate: Option[Instant] = None,
+                             traderName: String = "",
+                             goodName: String = "",
+                             entryNumber: String = "",
+                             traderCommodityCode: String = "",
+                             officerCommodityCode: String = "",
+                             contactName: String = "",
+                             contactEmail: Option[String] = None,
+                             contactPhone: String = "")
+
+object LiabilityFormData {
+
+  private type FormDate = (Int, Int, Int)
+
+  private val tupleToInstant: FormDate => Instant = {
+    case (day, month, year) =>
+      val min = LocalDate.of(year, month, day)
+      min.atStartOfDay(UTC).toInstant
+  }
+  private val instantToTuple: Instant => FormDate = { date =>
+    val offsetDate = date.atOffset(UTC).toLocalDate
+    (offsetDate.getDayOfMonth, offsetDate.getMonthValue, offsetDate.getYear)
+  }
+
+  private val validDateFormat: FormDate => Boolean = {
+    case (day, month, year) => Try(LocalDate.of(year, month, day)).isSuccess
+  }
+
+  private val emailRegex = """^[a-zA-Z0-9\.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$""".r
```

Comment:
Do we really have to validate email like this?  email regex is notoriously hard to get "right"

Reply:
TraderUI appears to be using org.apache.commons.validator.routines.EmailValidator.  Maybe we could do the same here?

Reply:
I think he copied this from the Scala Play email validation constraint

Reply:
Is either to introduce a new library and keep it updated just for this email validation... There is another Validation on play which checks the email in the same way you would check a text, number, etc... but because an empty is valid and we need our own error message we have created in this way.